pkg/util/log.go

```diff
@@ -103,10 +103,12 @@ func WithContext(ctx context.Context, l log.Logger) log.Logger {
 	// Weaveworks uses "orgs" and "orgID" to represent Cortex users,
 	// even though the code-base generally uses `userID` to refer to the same thing.
 	userID, err := user.ExtractOrgID(ctx)
-	if err != nil {
+	if err != nil && err != user.ErrNoOrgID {
 		return l
 	}
-	l = WithUserID(userID, l)
+	if err == nil {
+		l = WithUserID(userID, l)
+	}
```

Comment:
Can't we simplify it as the following, so that the trace ID is injected in any case?

```
if err == nil {
    l = WithUserID(userID, l)
}
```

Reply:
Agree, let's do that.